crates/distribution-types/src/index.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{IndexUrl, IndexUrlError};
+use std::str::FromStr;
+use thiserror::Error;
+use url::Url;
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
+pub struct Index {
+    /// The name of the index.
+    ///
+    /// Index names can be used to reference indexes elsewhere in the configuration. For example,
+    /// you can pin a package to a specific index by name:
+    ///
+    /// ```toml
+    /// [[tool.uv.index]]
+    /// name = "pytorch"
+    /// url = "https://download.pytorch.org/whl/cu121"
+    ///
+    /// [tool.uv.sources]
+    /// torch = { index = "pytorch" }
+    /// ```
+    pub name: Option<String>,
+    /// The URL of the index.
+    ///
+    /// Expects to receive a URL (e.g., `https://pypi.org/simple`) or a local path.
+    pub url: IndexUrl,
+    /// Mark the index as explicit.
+    ///
+    /// Explicit indexes will _only_ be used when explicitly enabled via a `[tool.uv.sources]`
+    /// definition, as in:
+    ///
+    /// ```toml
+    /// [[tool.uv.index]]
+    /// name = "pytorch"
+    /// url = "https://download.pytorch.org/whl/cu121"
+    /// explicit = true
+    ///
+    /// [tool.uv.sources]
+    /// torch = { index = "pytorch" }
+    /// ```
+    #[serde(default)]
+    pub explicit: bool,
+    /// Mark the index as the default index.
+    ///
+    /// By default, uv uses PyPI as the default index, such that even if additional indexes are
+    /// defined via `[[tool.uv.index]]`, PyPI will still be used as a fallback for packages that
+    /// aren't found elsewhere. To disable the PyPI default, set `default = true` on at least one
+    /// other index.
+    ///
+    /// Marking an index as default will move it to the front of the list of indexes, such that it
+    /// is given the highest priority when resolving packages.
```

Comment:
I thought it was the opposite?

Reply:
Yeah sorry, this comment looks stale. It is the opposite.